woocommerce/woocommerce
# Add lint config to blocks to lint imports

### Changes proposed in this Pull Request:

There is an issue with the way blocks is setup right now:

Normally in most projects these days you'll have some kind of import linting static analysis done, via eslint. But with TypeScript the eslint import rules tend to be redundant because the TS compiler picks up unresolved imports. **But** because blocks uses TypeScript only in a checking capacity but does not actually error with TypeScript right now, we get the worst of both. We don't get import checking from eslint and we don't get import checking from TypeScript.

Recently this lead to a bad import finding it's way into trunk without any CI check or build step notifying the cause. This is unacceptable and I expect it will happen again in future. We really should have our static analysis catch simple things like this.

This PR's measures are a stop gap measure made with time saving in mind:

1. We re-enable import rules that are generally disabled in typescript eslint
2. A couple of them report quite a few errors related to `export *` right now, I'd like to fix those but it feels out of scope of this immediate issue
3. We leave `import/named` rule as error. This would have caught the error we saw today that broke code in trunk.
4. As a result of enabling this I caught one instance of a bad type import that had been lying around a while.

### How to test the changes in this Pull Request:

1. Checkout this branch
2. Introduce a false named import into a ts file. e.g. you could replicate the import error seen today, in `plugins/woocommerce-blocks/assets/js/blocks/collection-filters/inner-blocks/active-filters/frontend.ts` you could add:

```
import { navigate } from '../../utils';
```
3. From blocks plugin folder run `pnpm lint:js` and see it reports an error for the wrong import.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add additional lint to woocommerce-blocks that ensures bad imports are not introduced.

#### Comment 

</details>
